tests/typesystem/test_object.py

```diff
@@ -99,6 +106,23 @@ class test_object_default():
     assert value['location'] == {'latitude': 0.0, 'longitude': 0.0}
 
 
+class test_optional_keys():
+    value = ObjectWithOptionalValues({
+        'name': 'steve'
+    })
+    assert value['name'] == 'steve'
+
+
+class test_optional_keys_are_still_validated():
+    with pytest.raises(exceptions.TypeSystemError) as exc:
+        value = ObjectWithOptionalValues({
+            'name': 'steve',
+            'some_optional_number': 'hello'
+        })
+    assert str(exc.value) \
+        == '{\'some_optional_number\': \'Must be a valid number.\'}'
```

Comment:
for clarity of code would not it be better this way? (79 columns)
```python
assert str(exc.value) == "{'some_optional_number': 'Must be a valid number.'}"
```

Reply:
+1 for all on the same line, the linter might error on double quotes though

Reply:
works for me 👍